src/components/side_panel/find_and_replace/find_and_replace_store.ts

```diff
@@ -110,10 +111,12 @@ export class FindAndReplaceStore extends SpreadsheetStore implements HighlightPr
   }
 
   finalize() {
-    if (this.isSearchDirty) {
+    if (this.isSearchDirty && !this.isAutoSheetSwitch) {
       this.refreshSearch(false);
       this.isSearchDirty = false;
     }
+    this.findMatches();
```

Comment:
`findMatches` is not necessarily fast if the spreadsheet is really large. It's not a good idea to do it for *every* command finalize.

I think I'd try to split `case "ACTIVATE_SHEET":` from the other cases and handle that there.